### PR TITLE
noeval fix, use proper destination pointer and don't assume word alig…

### PR DESCRIPTION
**Pull request by jean-m-cyr**
Fix issue #915

### CODE REVIEW

File: libethash-cuda/CUDAMiner.cpp [hunk 1]
```diff
@@ -508,17 +508,8 @@ void CUDAMiner::search(
...
-						m[1] = buffer->result[j].mix[1] ;
-						m[2] = buffer->result[j].mix[2] ;
-						m[3] = buffer->result[j].mix[3] ;
-						m[4] = buffer->result[j].mix[4] ;
-						m[5] = buffer->result[j].mix[5] ;
-						m[6] = buffer->result[j].mix[6] ;
-						m[7] = buffer->result[j].mix[7] ;
-					}
+					if (s_noeval)
+						memcpy(mixes[j].data(), (const void *)buffer->result[j].mix, sizeof(buffer->result[j].mix));
```
Comment (chfast) [inline libethash-cuda/CUDAMiner.cpp]: Just do `&` instead of `(const void*)`.

Comment (jean-m-cyr) [reply libethash-cuda/CUDAMiner.cpp]: buffer->result[j].mix is already a pointer (mix is an array), but it is of volatile * type

Wouldn't using an & here result in a pointer to a pointer?

Comment (chfast) [reply libethash-cuda/CUDAMiner.cpp]: The `.mix` in an array (auto-decay to pointer). It should work as 
```
memcpy(mixes[j].data(), &buffer->result[j].mix, sizeof(buffer->result[j].mix));
```
should be correct and looks better.

**Comment by jean-m-cyr** [comment]
memcpy(mixes[j].data(), &buffer->result[j].mix, sizeof(buffer->result[j].mix));
gets you: error: invalid conversion from ‘volatile void*’ to ‘const void*’

Both:
memcpy(mixes[j].data(), (void *)&buffer->result[j].mix, sizeof(buffer->result[j].mix));
and:
memcpy(mixes[j].data(), (void *)buffer->result[j].mix, sizeof(buffer->result[j].mix));
are equivalent, yield the same address to memcpy, and result in valid shares.

**Comment by dieuquocgia** [comment]
down hashrate

**Pull request merged**